cpu/fe310/include/irq_arch.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2017 Ken Rabold
+ *               2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup         cpu_fe310
+ * @{
+ *
+ * @file
+ * @brief           Implementation of the kernels irq interface
+ *
+ * @author          Ken Rabold
+ * @author          Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef IRQ_ARCH_H
+#define IRQ_ARCH_H
+
+#include <stdint.h>
+#include "irq.h"
+#include "cpu_conf.h"
+#include "cpu.h"
+
+#include "vendor/encoding.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern volatile int fe310_in_isr;
+
+/**
+ * @brief Enable all maskable interrupts
+ */
+static inline __attribute__((always_inline)) unsigned int irq_enable(void)
+{
+    /* Enable all interrupts */
+    unsigned retval = read_csr(mstatus);
+    set_csr(mstatus, MSTATUS_MIE);
+    return retval;
```

Comment:
```suggestion
    unsigned retval = read_csr(mstatus);
    set_csr(mstatus, MSTATUS_MIE);
    return retval;
    unsigned state;
    __asm__ volatile (
        "csrrs %[dest], mstatus, %[mask]"
        : [dest]    "=r"(state)
        : [mask]    "i"(MSTATUS_MIE)
        : "memory"
    );

    return state;
```

This should also contain the memory barrier (and make use of a single instruction instead of two). The code is identical to `irq_disable`, except of `csrrc` (CSR read and clear) being replaced with `csrrs` (CSR read and set).

Reply:
Even better with `tests/bench_mutex_pingpong`:

```
2020-06-26 10:06:28,946 # main starting
2020-06-26 10:06:29,948 # { "result" : 545096 }
```

I have the same results for the other bench.

In terms of code size, this PR save 48 bytes when building `examples/default`.